cloudcix_primitives/network_secondary_lxd.py

```diff
@@ -0,0 +1,249 @@
+                network_device = device.copy()
+                # Check if there's a MAC address in the config using the lxd.py convention
+                mac_key = f'volatile.{name}.hwaddr'
+                if mac_key in config:
+                    network_device['mac_address'] = config[mac_key]
+                network_devices[name] = network_device
+
+        if not network_devices:
+            # No secondary network interfaces found
+            result['network_devices'] = None
+            fmt.add_successful('network_devices.read', {'status': 'none_found'})
+            return True, f'1201: {messages[1201]}', fmt.successful_payloads, {endpoint_url: result}
+
+        # Return the network devices
+        result['network_devices'] = network_devices
+        fmt.add_successful('network_devices.read', {'status': 'found', 'count': len(network_devices)})
+        return True, f'1200: {messages[1200]}', fmt.successful_payloads, {endpoint_url: result}
+
+    status, msg, successful_payloads, result = run_host(endpoint_url, 3220, {})
+
+    if status is False:
+        return False, {}, msg
+
+    return True, result, msg
+
+def scrub(
+    endpoint_url: str,
+    project: str,
+    container_name: str,
+    device_name: str,
+    verify_lxd_certs: bool = True,
+) -> Tuple[bool, str]:
+    """
+    Remove a secondary network interface from an LXD container.
+    """
+    # Define the messages
+    messages = {
+        1100: f'Successfully removed network interface {device_name} from container {container_name} on {endpoint_url}',
+        3121: f'Failed to connect to {endpoint_url} for container operations',
+        3122: f'Failed to retrieve container {container_name}. Payload exited with status ',
+        3124: f'Network interface {device_name} does not exist on container {container_name}.',
+        3125: f'Failed to remove network interface: ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        # Get the container
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+
+        instance = ret['payload_message']
+        fmt.add_successful('containers.get', ret)
+
+        # Check if the device exists
+        devices = instance.devices
+
+        if device_name not in devices:
+            return False, f"{prefix+4}: {messages[prefix+4]}", fmt.successful_payloads
```

Comment:
Similar if the `device_name` is not in the devices for a scrub request, this can return True as it is the expected result